src/Sarif.Multitool/QueryCommand.cs

```diff
@@ -71,11 +75,14 @@ public int RunWithoutCatch(QueryOptions options)
                 {
                     foreach (Result result in run.Results)
                     {
-                        Console.WriteLine(result.FormatForVisualStudio(result.GetRule(run)));
+                        Console.WriteLine(result.FormatForVisualStudio());
                     }
                 }
             }
 
+            // Remove any Runs with no remaining matches
+            log.Runs = log.Runs.Where(r => (r?.Results?.Count ?? 0) > 0).ToList();
```

Comment:
>log.Runs [](start = 12, length = 8)

I think this is the right decision. In result matching, we wouldn't want to discard a run that was left with no results. That empty run indicates 'run succeeded and hooray, it appears everything was fixed.' here the empty runs resulting from a query simply would require unnecessary post-processing. Just wanted to think through whether there's a useful distinction between something being entirely absent vs. explicitly present and empty here. I don't think there is, interested in your take.

Reply:
Good point. In the query case, it's nice to filter empty runs out to reduce noise when reading the log or when debugging subsequent actions on the log. If a future query scenario makes us want to keep empty runs, we could expose an option at that time.